h5py/h5d.pyx

```diff
@@ -497,3 +503,92 @@ cdef class DatasetID(ObjectID):
                     H5Sclose(space_id)
 
             return filters, bytes(data)
+
+    IF HDF5_VERSION >= (1, 10, 5):
+
+        @with_phil
+        def get_num_chunks(self, SpaceID space=None):
+            """ (SpaceID space=None) => INT num_chunks
+
+            Retrieve the number of chunks that have nonempty intersection with a
+            specified dataspace. Currently, this function only gets the number
+            of all written chunks, regardless of the dataspace.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef hsize_t num_chunks
+
+            if space is None:
+                space = self.get_space()
+            H5Dget_num_chunks(self.id, space.id, &num_chunks)
+            return num_chunks
+
+        @with_phil
+        def get_chunk_info(self, hsize_t index, SpaceID space=None):
+            """ (hsize_t index, SpaceID space=None) => StoreInfo
+
+            Retrieve storage information about a chunk specified by its index.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef haddr_t byte_offset
+            cdef hsize_t size
+            cdef hsize_t *chunk_offset
+            cdef unsigned filter_mask
+            cdef hid_t space_id = 0
+            cdef int rank
+
+            if space is None:
+                space_id = H5Dget_space(self.id)
+            else:
+                space_id = space.id
+
+            rank = H5Sget_simple_extent_ndims(space_id)
+            chunk_offset = <hsize_t*>emalloc(sizeof(hsize_t) * rank)
+            H5Dget_chunk_info(self.id, space_id, index, chunk_offset,
+                              &filter_mask, &byte_offset, &size)
+            cdef tuple cot = convert_dims(chunk_offset, <hsize_t>rank)
+            efree(chunk_offset)
+
+            if space is None:
+                H5Sclose(space_id)
+
+            return StoreInfo(index,
+                             cot if byte_offset != HADDR_UNDEF else None,
+                             filter_mask,
+                             byte_offset if byte_offset != HADDR_UNDEF else None,
+                             size)
+
+
+        @with_phil
+        def get_chunk_info_by_coord(self, tuple chunk_offset not None):
+            """ (TUPLE chunk_offset) => StoreInfo
+
+            Retrieve information about a chunk specified by the array
+            address of one of its dataset elements.
```

Comment:
The documentation seems to say that the coordinate given needs to be the first in the dataset, not any.

Reply:
Fixed.

Reply:
Is there a push still to come? I was suggesting that the docstring should be tweaked slightly.

Reply:
My bad. I somehow managed to lose the new text when merging with master.

Reply:
Thanks, the new wording looks good.